src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,68 @@
+package duke.command;
+
+import duke.command.Command;
+import duke.main.Ui;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Utils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class DeleteCommand implements Command {
+
+    @Override
+    public void execute(String task, Ui ui, Storage storage, TaskList taskList) {
+        String[] token = task.split(" ", 2);
+        if (token.length < 2) {
+            ui.setResponse("Please specify which task to delete");
+            return;
+        }
+        String[] indices = token[1].split(" ");
```

Comment:
Depending on your personal preference for the level of Abstraction, may I suggest creating simple methods to handle small actions like these? Having simple methods with readible names would allow the future collaborators of your code to easily understand what you want to do, while also aids in a neater code.

Reply:
Adding on, the above would add much code resuability as the same code could be applied in different classes.
If you would like, you could change your Command Interface to an Abstract Class instead and have these subclasses extend from it. Then, you can apply the same methods with differing messages/purposes.